tests/plugins/test_resolve_composes.py

```diff
@@ -290,6 +290,34 @@ def test_request_compose_for_modules(self, workflow, reactor_config_map):
 
         self.run_plugin_with_args(workflow, reactor_config_map=reactor_config_map)
 
+    @pytest.mark.parametrize(('with_modules'), (True, False))
+    def test_request_compose_empty_packagese(self, workflow, reactor_config_map, with_modules):
+        repo_config = dedent("""\
+            compose:
+                packages:
+            """)
+        if with_modules:
+            repo_config = dedent("""\
+                compose:
+                    packages:
+                    modules:
+                    - spam_modules
+                    - bacon_modules
+                    - eggs_modules
```

Comment:
Oh, I see, and the modules will be ignored.

One thing we *could* do is adjust the schema so that either 'packages' or 'modules' is specified, but not both:
```
"compose": {
  ...
    "oneOf": [
    {
      "required": [
        "packages"
      ]
    },
    {
      "required": [
        "modules"
      ]
    },
  ]
}
```

Since providing both 'packages' and 'modules' satisfies both of these requirements, the "oneOf" specifier will fail:
```
<ValidationError: "{'packages': None, 'modules': ['spam_modules', 'bacon_modules', 'eggs_modules']} is valid under each of {u'required': [u'modules']}, {u'required': [u'packages']}"
```
What do you think about doing this (perhaps as a separate story)? @mlangsdorf @rcerven @MartinBasti

Reply:
The docs specify "The 'packages' and 'modules' keys are mutually exclusive. If both are
provided, 'modules' will be ignored."  So that's what I implemented.

If the preference is to implement failure on multiple provision, I'm fine with that and it's probably a more sensible solution.

Reply:
How about if I file a separate story to make them mutually exclusive such that providing both is an error, including updating the docs to state that? I think it would involve updating some test cases as well.